### PR TITLE
chore: Migrates cloud_backup_snapshot_export_bucket to new auto-generated SDK

### PR DESCRIPTION
**Pull request by EspenAlbert**
## Description

Migrates `cloud_backup_snapshot_export_bucket` to new auto-generated SDK

Link to any related issue(s): CLOUDP-246317

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

## Further comments

Detected problem with data source having `id` required. Will be handled in separate PR. [Jira Ticket](https://jira.mongodb.org/browse/CLOUDP-247721)

### CODE REVIEW

File: internal/service/cloudbackupsnapshotexportbucket/resource_cloud_backup_snapshot_export_bucket.go [hunk 1]
```diff
@@ -248,7 +249,7 @@ func resourceCloudBackupSnapshotExportBucketRefreshFunc(ctx context.Context, cli
 						return nil, "REPEATING", err
 					}
 
-					if s.ChangeStatus == matlas.ChangeStatusPending {
+					if s.GetChangeStatus() == "PENDING" {
```
Comment (EspenAlbert) [inline internal/service/cloudbackupsnapshotexportbucket/resource_cloud_backup_snapshot_export_bucket.go:249]: Couldn't find any enum/constant in the new SDK.
Is this the expected way? @lantoli

Comment (lantoli) [reply internal/service/cloudbackupsnapshotexportbucket/resource_cloud_backup_snapshot_export_bucket.go:249]: yeah, unfortunately that's the case in this situation.  with the former manual SDK some constants were created manually

File: internal/service/cloudbackupsnapshotexportbucket/resource_cloud_backup_snapshot_export_bucket.go [hunk 2]
```diff
@@ -62,43 +62,43 @@ func returnCloudBackupSnapshotExportBucketSchema() map[string]*schema.Schema {
...
+		"id":         bucketResponse.GetId(),
 	}))
 
-	return resourceMongoDBAtlasCloudBackupSnapshotExportBucketRead(ctx, d, meta)
+	return resourceRead(ctx, d, meta)
 }
 
-func resourceMongoDBAtlasCloudBackupSnapshotExportBucketRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+func resourceRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
 	// Get client connection.
```
Comment (lantoli) [inline internal/service/cloudbackupsnapshotexportbucket/resource_cloud_backup_snapshot_export_bucket.go]: comment can be deleted

File: internal/service/cloudbackupsnapshotexportbucket/resource_cloud_backup_snapshot_export_bucket.go [hunk 3]
```diff
@@ -62,43 +62,43 @@ func returnCloudBackupSnapshotExportBucketSchema() map[string]*schema.Schema {
...
-	conn := meta.(*config.MongoDBClient).Atlas
+	conn := meta.(*config.MongoDBClient).AtlasV2
 	ids := conversion.DecodeStateID(d.Id())
 	projectID := ids["project_id"]
 	bucketID := ids["id"]
 
-	exportBackup, _, err := conn.CloudProviderSnapshotExportBuckets.Get(ctx, projectID, bucketID)
+	exportBackup, _, err := conn.CloudBackupsApi.GetExportBucket(ctx, projectID, bucketID).Execute()
 	if err != nil {
 		// case 404
```
Comment (lantoli) [inline internal/service/cloudbackupsnapshotexportbucket/resource_cloud_backup_snapshot_export_bucket.go]: knit as in other cases: 
// case 404: deleted in the backend case

**Review by lantoli — APPROVED** [review]

**Review by lantoli — APPROVED** [review]

**Pull request merged**